src/components/namestable/nametable-section.component.tsx

```diff
@@ -1,11 +1,56 @@
-import React from "react";
+import React, { useContext, useEffect, useMemo, useState } from "react";
 import { DataRow } from "../../logic/schedule-logic/data-row";
+import { WorkerInfoComponent } from "./worker-info.component";
+import { Drawer } from "@material-ui/core";
+import { WorkerType } from "../../common-models/worker-info.model";
+import { Sections } from "../../logic/providers/schedule-provider.model";
+import { ScheduleLogicContext } from "../schedule-page/table/schedule/use-schedule-state";
+import { ShiftsInfoLogic } from "../../logic/schedule-logic/shifts-info.logic";
 
 export interface NameTableCellOptions {
   dataRow: DataRow[];
+  workerType: WorkerType;
 }
 
-export function NameTableSection({ dataRow }: NameTableCellOptions): JSX.Element {
+export interface WorkerInfo {
+  name?: string;
+  workerType?: WorkerType;
+  requiredHours?: number;
+  actualHours?: number;
+  overtime?: number;
+  sumOfHours?: number;
+}
+
+export interface ToggleOpenState {
+  open: boolean;
+  name: string;
+}
+
+const workerInfo: WorkerInfo = { name: "" };
+
+export function NameTableSection({ dataRow, workerType }: NameTableCellOptions): JSX.Element {
+  const [openState, setOpen] = useState<ToggleOpenState>();
+
+  const scheduleLogic = useContext(ScheduleLogicContext);
+  const sectionKey: keyof Sections =
+    workerType === WorkerType.NURSE ? "NurseInfo" : "BabysitterInfo";
+  const shiftLogic = scheduleLogic?.getSection<ShiftsInfoLogic>(sectionKey);
+  console.log(shiftLogic);
+  function toggleDrawer(openState: ToggleOpenState): void {
+    setOpen(openState);
+    if (openState.open) {
+      const [requiredHours, actualHours, overtime] =
+        shiftLogic?.calculateWorkerHourInfo(openState.name) ?? [];
+
+      workerInfo.name = openState.name;
+      workerInfo.workerType = workerType;
+      workerInfo.requiredHours = requiredHours;
+      workerInfo.actualHours = actualHours;
+      workerInfo.overtime = overtime;
+      workerInfo.sumOfHours = actualHours + overtime;
```

Comment:
Nie potrzebujesz tego. 
`actualHour` - to wypracowane godziny. 
`Overtime` - jest zdefiniowane jako `actualHours` -  norma na miejsiąc.

Jeżeli chciałeś tutaj pokazać ile godzin pracował pracownik, to powinieneś użyć `actualHours`